src/gurt/misc.c

```diff
@@ -1214,40 +1211,68 @@ d_getenv_char(const char *name, char *char_val)
 }
 
 static int
-d_getenv_ull(unsigned long long *val, const char *name)
+d_getenv_ull(unsigned long long *val, const char *name, size_t val_size)
 {
 	char              *env;
+	char              *env_tmp = NULL;
 	char              *endptr;
-	unsigned long long tmp;
+	unsigned long long val_tmp;
 	int                rc;
 
 	assert(val != NULL);
 	assert(name != NULL);
+	assert(val_size <= sizeof(unsigned long long));
 
 	d_env_rwlock_rdlock();
 	env = getenv(name);
 	if (env == NULL) {
 		rc = -DER_NONEXIST;
+		d_env_rwlock_unlock();
 		goto out;
 	}
 
-	if (!dis_unsigned_str(env)) {
-		rc = -DER_INVAL;
+	/* DAOS-14896 NOTES:
+	 * - Duplicate env to reduce data race condition with external libraries not using the DAOS
+	 *   thread safe environment variables management API.
+	 * - Use of strdup() as there is no limit to environment variable size.
+	 */
+	env_tmp = strdup(env);
+	if (env_tmp == NULL) {
+		rc = -DER_NOMEM;
+		d_env_rwlock_unlock();
 		goto out;
 	}
+	d_env_rwlock_unlock();
 
-	errno = 0;
-	tmp   = strtoull(env, &endptr, 0);
-	if (errno != 0 || endptr == env || *endptr != '\0') {
+	errno   = 0;
+	val_tmp = strtoull(env_tmp, &endptr, 10);
```

Comment:
[Question] why change to decimal?

Reply:
Mostly to be consistent with the initial function which were only supporting decimal value.